include/internal/libfreenect2/async_packet_processor.h

```diff
@@ -119,6 +135,7 @@ class AsyncPacketProcessor : public PacketProcessor<PacketT>
       {
         // invoke process impl
         processor_->process(current_packet_);
+        releaseBuffer(current_packet_);
```

Comment:
why do you call releaseBuffer here? shouldn't the stream parser do this? otherwise you assume that every packetprocessor is wrapped with an asyncpacketprocessor, right?